newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -131,227 +79,200 @@ type Props = {|
-
-                    // Error handling:
-                    error: !!props.errorText,
-                    helperText: props.errorText || helperText,
-
-                    // Display:
-                    InputLabelProps: getLabelProps({ shrink: true }),
-
-                    // Events:
-                    InputProps: {
-                      // We wrap the onBlur/onFocus as we're a "semi controlled" field
-                      onBlur: wrappedOnBlur,
-                      onFocus: onFocus,
-                    },
-
-                    // Props for the input field from downshift:
-                    inputProps,
-
-                    // Style:
-                    style: props.textFieldStyle,
-                    fullWidth: props.fullWidth,
-                    ...computeTextFieldStyleProps(props),
-
-                    inputRef: this._input,
-                  })}
-                  <Popper
-                    open={isOpen}
-                    anchorEl={this._input.current}
-                    style={styles.menuPopper}
-                  >
-                    <div
-                      {...(isOpen
-                        ? getMenuProps({}, { suppressRefError: true })
-                        : {})}
-                    >
-                      {isOpen ? (
-                        <Paper
-                          style={{
-                            ...styles.menuPaper,
-                            width: this._input.current
-                              ? this._input.current.clientWidth
-                              : undefined,
-                          }}
-                          square
-                        >
-                          {filterDataSource(props.dataSource, inputValue).map(
-                            (item, index) =>
-                              renderItem({
-                                item,
-                                index,
-                                menuItemProps: getItemProps({ item, index }),
-                                highlightedIndex,
-                                selected: item === selectedItem,
-                              })
-                          )}
-                        </Paper>
-                      ) : null}
-                    </div>
-                  </Popper>
-                </div>
+                  {...other}
+                  {...computeTextFieldStyleProps(this.props)}
+                  style={this.props.textFieldStyle}
+                  label={this.props.floatingLabelText}
+                  inputRef={this._input}
+                  disabled={this.props.disabled}
+                  error={!!this.props.errorText}
+                  helperText={helperText}
+                  fullWidth={this.props.fullWidth}
+                />
               );
             }}
-          </Downshift>
+          />
         )}
       </I18n>
     );
   }
 }
+
+export default withStyles(themeStyles)(SemiControlledAutoComplete);
```

Comment:
Not a big fan of the High Order Component version - it breaks the static typing it seems. Prefer to use the hook API: https://material-ui.com/styles/basics/#hook-api

(makeStyle and useStyle).

Reply:
Oh you're back at having a class component.. in this case, can we use the "styled-component" API? https://material-ui.com/styles/basics/#styled-components-api

=> Make a `StyledAutocomplete` component, then use it instead of `Autocomplete` :) This will also shorten slightly the render function.

(or see if it's easy to turn the component into a function component and use the hook API)

Reply:
Okay, I'd refactor the code then. I used HOC only because I didn't want to convert the class component into a function. 😅

Reply:
In the long run that would be great! Otherwise take a look at this API https://material-ui.com/styles/basics/#styled-components-api - but if we can go ahead with a function that's even better! :)